File: src/main/resources/static/js/services/skillLevel.service.js
app.service('skillLevelServices', function($http) {
	var that = this;
	that.baseUrl = "/skill_level";

	that.getAll = function() {
		return $http.get(that.baseUrl + "/");
	};

	that.get = function(id) {
		return $http.get(that.baseUrl + "/" + id);
	};

	that.update = function(id, model) {
		return $http.put(that.baseUrl + "/" + id, model);
	};

	that.remove = function(id) {
		return $http.get(that.baseUrl + "/delete/" + id);
	};

	that.save = function(model) {
		console.log("levelService");
		console.log(model);
		return $http.post(that.baseUrl + "/save", model);
	};
});

File: src/main/java/com/ymagis/annualappraisal/service/SkillLevelService.java
package com.ymagis.annualappraisal.service;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.logging.Level;
import java.util.logging.Logger;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.ymagis.annualappraisal.model.SkillLevel;
import com.ymagis.annualappraisal.repository.SkillLevelRepository;

@Service(value = "skillLevelService")
public class SkillLevelService {

	private static Logger logger = Logger.getLogger(SkillLevelService.class.getName());

	@Autowired
	private SkillLevelRepository repository;

	public List<SkillLevel> getAll() {
		try {
			return repository.findAll();
		} catch (Exception e) {
			logger.log(Level.SEVERE, e.getMessage(), e);
		}
		return new ArrayList<>();
	}

	public SkillLevel save(SkillLevel skillLevel) {
		try {
			repository.save(skillLevel);
		} catch (Exception e) {
			logger.log(Level.SEVERE, e.getMessage(), e);
		}
		return skillLevel;
	}

	public SkillLevel get(Long id) {
		try {
			Optional<SkillLevel> item = repository.findById(id);
			if (item.isPresent()) {
				return item.get();
			}
		} catch (Exception e) {
			logger.log(Level.SEVERE, e.getMessage(), e);
		}
		return null;
	}

	public void delete(Long id) {
		try {
			Optional<SkillLevel> item = repository.findById(id);
			if (item.isPresent()) {
				repository.deleteById(id);
			}
		} catch (Exception e) {
			logger.log(Level.SEVERE, e.getMessage(), e);
		}
	}
}
